src/manual/media-recorder/js/main.js

```diff
@@ -0,0 +1,441 @@
+*  tree.
+*/
+
+'use strict';
+
+function RecorderTest() {
+  this.theLocalStream = null;
+  this.theRemoteStream = null;
+  this.theRecorder = null;
+  this.eventsMonitored = null;
+  // Note: ondataavailable is not in the list below because it is necessary to
+  // have it always on if we ever want to playbabk/download the recording in a
+  // consistent manner.
+  this.RECORDER_EVENTS = ['onstart', 'onpause', 'onresume', 'onstop',
+      'onerror'];
+  this.eventCounter = 0;
+  this.recordedData = {};
+  this.timeDataReceived = null;
+}
+
+RecorderTest.prototype = {
+  // This function sets up the different event handlers necessary to run the
+  // ui.
+  setup: function() {
+    this.onButtonClick('GetUserMedia', this.getUserMedia.bind(this));
+    this.onButtonClick('CreatePeerConnection', this.createPeerConnection.bind(
+        this));
+    this.onButtonClick('CreateRecorder', this.createRecorder.bind(this));
+    this.onButtonClick('Start', this.startRecorder.bind(this));
+    this.onButtonClick('Pause', this.pauseRecorder.bind(this));
+    this.onButtonClick('Resume', this.resumeRecorder.bind(this));
+    this.onButtonClick('Stop', this.stopRecorder.bind(this));
+    this.onButtonClick('RequestData', this.requestDataRecorder.bind(this));
+    this.onButtonClick('Playback', this.playback.bind(this));
+    this.onButtonClick('Download', this.download.bind(this));
+    var monitorSelect = document.getElementById('monitorEvents');
+    monitorSelect.onchange = this.monitorRecorderEvents.bind(this, event);
+  },
+
+  // This function creates a button onClick event handler.
+  onButtonClick: function(buttonName, callBack) {
+    console.log('button name', buttonName);
+    document.getElementsByName(buttonName)[0].addEventListener('click',
+        callBack);
+  },
+
+  // This function configures the event handlers of media recorder.
+  monitorRecorderEvents: function(event) {
+    console.log('monitor debug:', event);
+    if (!this.theRecorder) {
+      alert('No recorder initialized!');
+      return;
+    }
+    if (this.eventsMonitored) {
+      this.recorderEventHandlersAction('remove', this.eventsMonitored);
+    }
+    var eventName = event.path[0].value;
+    if (eventName) {
+      this.recorderEventHandlersAction('add', eventName);
+    }
+    this.eventsMonitored = eventName;
+  },
+
+  // This function gets the selected value of a <select> element.
+  getSelectedValue: function(id) {
+    var e = document.getElementById(id);
+    return e.options[e.selectedIndex].value;
+  },
+
+  // This function builds the basic gUM constraints from the
+  // selected GetUserMedia options in html.
+  buildConstraints: function(audioEnabled, videoEnabled, resolution) {
+    var constraints = {audio: false, video: false};
+    var res = resolution.split('x');
+    var width = res[0];
+    var height = res[1];
+    if (height && videoEnabled) {
+      constraints.video = {mandatory: {
```

Comment:
This is the old constraints style, see this [sample page](https://github.com/webrtc/samples/blob/gh-pages/src/content/getusermedia/resolution/js/main.js) for what the new ones look like.